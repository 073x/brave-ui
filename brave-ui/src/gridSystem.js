/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import React, { PureComponent } from 'react'
import { StyleSheet, css } from 'aphrodite/no-important'

// Create a 12-based column grid system
export const gridTemplate = 12

export class Grid extends PureComponent {
  get componentStyles () {
    const { props } = this
    const customStyle = {}

    if ('padding' in props) {
      customStyle['--gridPadding'] = props.padding
    }

    if ('gap' in props) {
      customStyle['--gridGap'] = props.gap
    }

    if ('width' in props) {
      customStyle['--gridWidth'] = props.width
    }

    if ('height' in props) {
      customStyle['--gridHeight'] = props.height
    }

    if ('textColor' in props) {
      customStyle['--gridTextColor'] = props.textColor
    }

    if ('background' in props) {
      customStyle['--gridBackground'] = props.background
    }

    return customStyle
  }

  render () {
    const { id, disabled, children, style } = this.props
    return (
      <div
        id={id}
        className={css(styles.grid, disabled && styles.grid__disabled)}
        style={Object.assign(this.componentStyles, style)}>
        {children}
      </div>
    )
  }
}

const gridWrapper = {
  grid: {
    boxSizing: 'border-box',
    display: 'grid',
    gridTemplateColumns: `repeat(${gridTemplate}, 1fr)`,
    gridGap: 'var(--gridGap, 15px)',
    padding: 'var(--gridPadding, 0)',
    maxWidth: 'var(--gridWidth)',
    height: 'var(--gridHeight)',
    color: 'var(--gridTextColor)',
    backgroundColor: 'var(--gridBackground)'
  },

  grid__disabled: {
    opacity: 0.3,
    userSelect: 'none'
  }
}

export class Column extends PureComponent {
  get componentStyles () {
    const customStyle = {}
    const { props } = this
    const needsFlex = 'align' in props || 'verticalAlign' in props

    if (needsFlex) {
      customStyle['--columnDisplay'] = 'flex'
    }

    if ('align' in props) {
      customStyle['--columnAlign'] = props.align
    }

    if ('verticalAlign' in props) {
      customStyle['--columnVerticalAlign'] = props.verticalAlign
    }

    if ('background' in props) {
      customStyle['--columnBackground'] = props.background
    }

    return customStyle
  }

  render () {
    const { id, size = 12, children } = this.props
    return (
      <div
        id={id}
        className={css(styles[`column${size}`])}
        style={this.componentStyles}>
        {children}
      </div>
    )
  }
}

// Iterate over gridTemplate and add a `columnN` classname
// i.e. {column1, ..., columnN}. Being N the max gridTemplate number
let columnSystem = {}
Array.from({length: gridTemplate}, (v, i) => i + 1).forEach((size) => {
  Object.assign(columnSystem, {
    [`column${size}`]: {
      boxSizing: 'border-box',
      position: 'relative',
      gridColumn: `span ${size}`,
      display: 'var(--columnDisplay, inherit)',
      justifyContent: 'var(--columnAlign)',
      alignItems: 'var(--columnVerticalAlign)',
      backgroundColor: 'var(--columnBackground)'
    }
  })
})

// merge both grid and column styles obj to be used in Aphrodite
const gridSystem = Object.assign(gridWrapper, columnSystem)
const styles = StyleSheet.create(gridSystem)
