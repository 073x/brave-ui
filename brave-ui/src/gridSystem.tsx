/* This Source Code Form is subject to the terms of the Mozilla Public
* License, v. 2.0. If a copy of the MPL was not distributed with this file,
* You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import { StyleSheet, css } from 'aphrodite/no-important'

// Create a 12-based column grid system
export const gridTemplate = 12
export interface GridProps {
  id?: string,
  disabled?: boolean,
  children?: React.ReactNode,
  // Component styles
  columns?: string,
  padding?: string,
  gap?: string,
  width?: string,
  height?: string,
  textColor?: string,
  background?: string
}

interface AphroditeStyle {
  [selector: string]: string | number | AphroditeStyle
}

class Grid extends React.PureComponent<GridProps, {}> {
  get componentStyles () {
    const { columns, padding, gap, width, height, textColor, background } = this.props
    return {
      '--gridTemplateColumns': columns || gridTemplate,
      '--gridPadding': padding,
      '--gridGap': gap,
      '--gridWidth': width,
      '--gridHeight': height,
      '--gridTextColor': textColor,
      '--gridBackground': background
    }
  }

  render () {
    const { id, disabled, children } = this.props
    return (
      <div
        id={id}
        className={css(styles.grid, disabled && styles.grid__disabled)}
        style={this.componentStyles}>
        {children}
      </div>
    )
  }
}

const gridWrapper = {
  grid: {
    boxSizing: 'border-box',
    display: 'grid',
    gridTemplateColumns: `repeat(var(--gridTemplateColumns), 1fr)`,
    gridGap: 'var(--gridGap, 15px)',
    padding: 'var(--gridPadding, 0)',
    maxWidth: 'var(--gridWidth)',
    height: 'var(--gridHeight)',
    color: 'var(--gridTextColor)',
    backgroundColor: 'var(--gridBackground)'
  },

  grid__disabled: {
    opacity: 0.3,
    userSelect: 'none'
  }
}

export interface ColumnProps {
  id?: string,
  size?: number,
  children?: React.ReactNode,
  // Component styles
  align?: string,
  verticalAlign?: string,
  background?: string,
  direction?: string
}

class Column extends React.PureComponent<ColumnProps, {}> {
  get componentStyles () {
    const { props } = this
    const needsFlex = (
      'align' in props ||
      'verticalAlign' in props ||
      'direction' in props
    )
    return {
      '--columnDisplay': needsFlex && 'flex',
      '--columnAlign': props.align,
      '--columnVerticalAlign': props.verticalAlign,
      '--columnBackground': props.background,
      '--columnDirection': props.direction
    }
  }

  render () {
    const { id, size = gridTemplate, children } = this.props
    return (
      <div
        id={id}
        className={css(styles[`column${size}`])}
        style={this.componentStyles}>
        {children}
      </div>
    )
  }
}

// Iterate over gridTemplate and add a `columnN` classname
// i.e. {column1, ..., columnN}. Being N the max gridTemplate number
let columnSystem = {}
Array.from({length: gridTemplate}, (v, i) => i + 1).forEach(size => {
  Object.assign(columnSystem, {
    [`column${size}`]: {
      boxSizing: 'border-box',
      position: 'relative',
      gridColumn: `span ${size}`,
      display: 'var(--columnDisplay, inherit)',
      justifyContent: 'var(--columnAlign)',
      alignItems: 'var(--columnVerticalAlign)',
      backgroundColor: 'var(--columnBackground)',
      flexDirection: 'var(--columnDirection)'
    }
  })
})

const gridSystem = Object.assign(gridWrapper, columnSystem)
const styles: AphroditeStyle = StyleSheet.create(gridSystem)

export { Grid, Column }
