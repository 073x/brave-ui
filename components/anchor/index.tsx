/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import './style.css'

import { applyClass } from '../helpers'

export interface ActionButtonProps {
  id?: string,
  href: string
  noStyle?: boolean
  target?: '_blank' |Â '_parent' | '_self' | '_top'
  text: string | number
}

class Anchor extends React.PureComponent<ActionButtonProps, {}> {
  render () {
    const { id, href, noStyle = false, target, text } = this.props
    return (
      <a
        id={id}
        className={applyClass({
          anchor: true,
          anchor__noStyle: noStyle
        })}
        href={href}
        target={target}
        rel='noreferrer noopener'
      >
        {text.toString()}
      </a>
    )
  }
}

export default Anchor
