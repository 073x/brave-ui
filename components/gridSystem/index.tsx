/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import * as React from 'react'
import './style.css'

import { applyClass } from '../helpers'

// Create a 12-based column grid system
export const gridTemplate = 12
export interface GridProps {
  id?: string,
  disabled?: boolean,
  children?: React.ReactNode,
  // Component styles
  columns?: string,
  padding?: string,
  gap?: string,
  width?: string,
  height?: string,
  textColor?: string,
  background?: string
}

class Grid extends React.PureComponent<GridProps, {}> {
  get componentStyles () {
    const { columns, padding, gap, width, height, textColor, background } = this.props
    return {
      '--gridTemplateColumns': columns != null ? columns : gridTemplate,
      '--gridPadding': padding,
      '--gridGap': gap,
      '--gridWidth': width,
      '--gridHeight': height,
      '--gridTextColor': textColor,
      '--gridBackground': background
    }
  }

  render () {
    const { id, disabled = false, children } = this.props
    return (
      <div
        id={id}
        className={applyClass({
          grid: true,
          grid__disabled: disabled
        })}
        style={this.componentStyles}>
        {children}
      </div>
    )
  }
}

export interface ColumnProps {
  id?: string,
  children?: React.ReactNode,
  // Component styles
  size?: number,
  align?: string,
  verticalAlign?: string,
  background?: string,
  direction?: string
}

class Column extends React.PureComponent<ColumnProps, {}> {
  get componentStyles () {
    const { props } = this
    const needsFlex = (
      'align' in props ||
      'verticalAlign' in props ||
      'direction' in props
    )
    return {
      '--columnSize': 'size' in props ? `span ${props.size}` : `span ${gridTemplate}`,
      '--columnDisplay': needsFlex && 'flex',
      '--columnAlign': props.align,
      '--columnVerticalAlign': props.verticalAlign,
      '--columnBackground': props.background,
      '--columnDirection': props.direction
    }
  }

  render () {
    const { id, children } = this.props
    return (
      <div
        id={id}
        className='column'
        style={this.componentStyles}>
        {children}
      </div>
    )
  }
}

export { Grid, Column }
