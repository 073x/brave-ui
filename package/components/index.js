"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const buttonPrimary_1 = require("./buttonsIndicators/buttonPrimary");
exports.ButtonPrimary = buttonPrimary_1.default;
const toggle_1 = require("./formControls/toggle");
exports.Toggle = toggle_1.default;
const buttonSecondary_1 = require("./buttonsIndicators/buttonSecondary");
exports.ButtonSecondary = buttonSecondary_1.default;
const modal_1 = require("./popupModals/modal");
exports.Modal = modal_1.default;
const table_1 = require("./dataTables/table");
exports.Table = table_1.default;
const checkbox_1 = require("./formControls/checkbox");
exports.Checkbox = checkbox_1.default;
const tabs_1 = require("./layout/tabs");
exports.Tabs = tabs_1.default;
const gridList_1 = require("./layout/gridList");
exports.Grid = gridList_1.Grid;
exports.Column = gridList_1.Column;
const buttonCta_1 = require("./buttonsIndicators/buttonCta");
exports.ButtonCta = buttonCta_1.default;
const textarea_1 = require("./formControls/textarea");
exports.TextArea = textarea_1.default;
const buttonGhost_1 = require("./buttonsIndicators/buttonGhost");
exports.ButtonGhost = buttonGhost_1.default;
const select_1 = require("./formControls/select");
exports.Select = select_1.default;
//# sourceMappingURL=index.js.map