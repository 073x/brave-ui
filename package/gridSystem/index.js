"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const style_1 = require("./style");
class Grid extends React.PureComponent {
    render() {
        return (React.createElement(style_1.StyledGrid, { id: this.props.id, disabled: this.props.disabled ? this.props.disabled : false, columns: this.props.columns, padding: this.props.padding, gap: this.props.gap, width: this.props.width, height: this.props.height, textColor: this.props.textColor, background: this.props.background }, this.props.children));
    }
}
exports.Grid = Grid;
class Column extends React.PureComponent {
    render() {
        return (React.createElement(style_1.StyledColumn, { id: this.props.id, size: this.props.size, align: this.props.align, verticalAlign: this.props.verticalAlign, background: this.props.background, direction: this.props.direction }, this.props.children));
    }
}
exports.Column = Column;
//# sourceMappingURL=index.js.map