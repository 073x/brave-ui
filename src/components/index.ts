/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

import Button from './buttonsIndicators/button'
import Checkbox from './formControls/checkbox'
import ControlWrapper from './formControls/controlWrapper'
import Input from './formControls/input'
import Modal from './popupModals/modal'
import Select from './formControls/select'
import Table from './dataTables/table'
import Tabs from './layout/tabs'
import Card from './layout/card'
import TextArea from './formControls/textarea'
import Toggle from './formControls/toggle'
import { Grid, Column } from './layout/gridList'
import Heading from './text/heading'

export {
  Button,
  Checkbox,
  Column,
  ControlWrapper,
  Grid,
  Heading,
  Input,
  Modal,
  Select,
  Table,
  Tabs,
  Card,
  TextArea,
  Toggle
}
