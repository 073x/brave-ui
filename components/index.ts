/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

 import TextLabel from './textLabel'
 import Anchor from './anchor'
 import UnstyledButton from './unstyledButton'
 import PushButton from './pushButton'
 import SwitchButton from './switchButton'
 import SelectOption from './selectOption'
 import { Grid, Column } from './gridSystem'
 import ContentToggleArrow from './contentToggleArrow'
 import Separator from './separator'
 import BoxedContent from './boxedContent'
 import Paragraph from './paragraph'
 import MediaContent from './mediaContent'

 export {
   TextLabel,
   Anchor,
   UnstyledButton,
   PushButton,
   SwitchButton,
   SelectOption,
   Grid,
   Column,
   ContentToggleArrow,
   Separator,
   BoxedContent,
   Paragraph,
   MediaContent
 }
