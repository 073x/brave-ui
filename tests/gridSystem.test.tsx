/* global expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import { Grid, Column } from '../components/gridSystem'

describe('gridSystem tests', () => {
  describe('grid component', () => {
    it('matches the snapshot', () => {
      const component = <Grid />
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(<Grid id='grid' />)
      const assertion = wrapper.find('#grid').length
      expect(assertion).toBe(1)
    })

    it('defines a custom padding', () => {
      const wrapper = shallow(<Grid padding='20px' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--gridPadding': '20px' })
      )
    })

    it('defines a custom gap', () => {
      const wrapper = shallow(<Grid gap='10px' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--gridGap': '10px' })
      )
    })

    it('defines a custom width', () => {
      const wrapper = shallow(<Grid width='13px' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--gridWidth': '13px' })
      )
    })

    it('defines a custom background', () => {
      const wrapper = shallow(<Grid background='cyan' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--gridBackground': 'cyan' })
      )
    })
  })

  describe('column component', () => {
    it('matches the snapshot', () => {
      const component = <Column />
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(<Column id='column' />)
      const assertion = wrapper.find('#column').length
      expect(assertion).toBe(1)
    })

    it('defines a custom align', () => {
      const wrapper = shallow(<Column align='flex-start' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--columnAlign': 'flex-start' })
      )
    })

    it('defines a custom verticalAlign', () => {
      const wrapper = shallow(<Column verticalAlign='flex-end' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--columnVerticalAlign': 'flex-end' })
      )
    })

    it('defines a custom background', () => {
      const wrapper = shallow(<Column background='purple' />)
      const assertion = wrapper.find('div').props().style
      expect(assertion).toEqual(
        expect.objectContaining({ '--columnBackground': 'purple' })
      )
    })
  })
})
