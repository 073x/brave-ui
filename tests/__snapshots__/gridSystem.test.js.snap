// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gridSystem tests column component matches the snapshot 1`] = `
<div
  className="column12_1wuh2fy"
  id={undefined}
  style={Object {}}
/>
`;

exports[`gridSystem tests grid component matches the snapshot 1`] = `
<div
  className="grid_7oehqz"
  id={undefined}
  style={Object {}}
/>
`;
